r1cs/gadgets/std/src/groups/curves/short_weierstrass/mnt/mnt6/mod.rs

```diff
@@ -143,30 +159,45 @@ impl<P: MNT6Parameters>G2PreparedGadget<P> {
         Ok(g2p)
     }
 
+    /* computes the preparation coefficients at S and its subsequent doubling
+    */
     fn doubling_step_for_flipped_miller_loop<CS: ConstraintSystem<P::Fp>>(
         mut cs: CS,
         s: &G2Gadget<P>,
     ) -> Result<(G2Gadget<P>, G2CoefficientsGadget<P>), SynthesisError>
     {
-        //Compute gamma
-        let three_sx_squared_plus_a = Fp3G::<P>::alloc(cs.ns(|| "3s_x^2 + a"), || {
+        /*
+          CAUTION
```

Comment:
You should remove this comment and merge with the last development (so that your fix will be here too) before approving the PR